### PR TITLE
Webhook bugfix and testworld webhook tests

### LINKED ISSUE #511: Integration testing for webhooks
We should test our webhooks with an integration test by setting up an internal http server that can listen to hooks.

### PR DESCRIPTION
**Pull request by vimukthi-git**
<This is intended as a guide when describing-reviewing pull requests for go-centrifuge>
closes #511

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,7 +45,7 @@ sudo apt-get -y upgrade
 wget https://dl.google.com/go/go1.11.4.linux-amd64.tar.gz
 
 #extract the aechive and move it to /usr/local folder
```
Comment (mikiquantum) [inline README.md]: Now that you are at it:
```suggestion
#extract the archive and move it to /usr/local folder
```

File: cmd/centrifuge/create_config.go [hunk 2]
```diff
@@ -41,7 +41,7 @@ func init() {
 				log.Error(err)
 			}
 
-			err = cmd.CreateConfig(targetDataDir, ethNodeURL, accountKeyPath, string(pwd), network, apiHost, apiPort, p2pPort, bootstraps, txPoolAccess, false, "", nil)
+			err = cmd.CreateConfig(targetDataDir, ethNodeURL, accountKeyPath, string(pwd), network, apiHost, apiPort, p2pPort, bootstraps, txPoolAccess, false, "", nil, "")
```
Comment (mikiquantum) [inline cmd/centrifuge/create_config.go]: Do you want to enable here as well, reading the notification_endpoint extra parameter?

File: config/configuration.go [hunk 3]
```diff
@@ -532,6 +532,10 @@ func CreateConfigFile(args map[string]interface{}) (*viper.Viper, error) {
 	txPoolAccess := args["txpoolaccess"].(bool)
 	preCommitEnabled := args["preCommitEnabled"].(bool)
 	apiHost := args["apiHost"].(string)
+	webhookURL, ok := args["webhookURL"].(string)
```
Comment (mikiquantum) [inline config/configuration.go]: I think we can simplify this to:
```suggestion
	webhookURL, _ := args["webhookURL"].(string)
```
or even without the `_` if we pass it always as the rest of the fields.

File: documents/coredocument.go [hunk 4]
```diff
@@ -56,23 +56,30 @@ const (
...
-	CDTreePrefix:         {1, 0, 0, 0},
-	SigningTreePrefix:    {2, 0, 0, 0},
-	SignaturesTreePrefix: {3, 0, 0, 0},
-	DRTreePrefix:         {4, 0, 0, 0},
-}
-
 // CompactProperties returns the compact property for a given prefix
 func CompactProperties(key string) []byte {
-	return compactProps[key]
+	switch key {
```
Comment (mikiquantum) [inline documents/coredocument.go]: 👍

File: jobs/jobsv1/manager.go [hunk 5]
```diff
@@ -123,9 +123,14 @@ func (s *manager) ExecuteWithinJob(ctx context.Context, accountID identity.DID,
...
-		done <- true
+		// non blocking send
+		select {
+		case done <- true:
+		default:
+			log.Debug("job done channel has no listener")
+		}
 
-		if mJob != nil {
+		if mJob != nil && jobs.JobIDEqual(existingJobID, jobs.NilJobID()) {
```
Comment (mikiquantum) [inline jobs/jobsv1/manager.go]: Just sent on the parent job?

File: cmd/centrifuge/create_config.go (hunk 2 above)
Comment (vimukthi-git) [reply cmd/centrifuge/create_config.go]: I would like to keep the changes to a minimum, We can do it in a future pr that makes changes to commands.

File: jobs/jobsv1/manager.go (hunk 5 above)
Comment (vimukthi-git) [reply jobs/jobsv1/manager.go]: yup, as child jobs are irrelevant to the users, as well as can explode the number of webhooks needlessly.

**Pull request merged**